pkg/lib/serverTime.js

```diff
@@ -321,13 +309,89 @@
-        const text = `# This file is automatically generated by Cockpit\n\n[Time]\n${enabled ? "" : "#"}NTP=${servers.join(" ")}\n`;
+        const text = `# This file is automatically generated by Cockpit\n\n[Time]\n${config.enabled ? "" : "#"}NTP=${config.servers.join(" ")}\n`;
 
         return cockpit.spawn(["mkdir", "-p", "/etc/systemd/timesyncd.conf.d"], { superuser: "try" })
                 .then(() => custom_ntp_config_file.replace(text));
+    }
+
+    const chronyd_sourcedir = "/etc/chrony/sources.d";
+    const chronyd_sources_enabled = chronyd_sourcedir + "/cockpit.sources";
+    const chronyd_sources_disabled = chronyd_sourcedir + "/cockpit.disabled";
+
+    function get_custom_ntp_chronyd() {
+        const enabled_file = cockpit.file(chronyd_sources_enabled, { superuser: "try" });
+        const disabled_file = cockpit.file(chronyd_sources_disabled, { superuser: "try" });
+
+        function parse_servers(data) {
+            if (!data)
+                return [];
+            const servers = [];
+            data.split("\n").forEach(function(line) {
+                const parts = line.split(" ");
+                if (parts[0] == "server")
+                    servers.push(parts[1]);
+            });
+            return servers;
+        }
+
+        return enabled_file.read()
+                .then(data => {
+                    if (data) {
+                        return {
+                            backend: "chronyd",
+                            enabled: true,
+                            servers: parse_servers(data)
+                        };
+                    } else {
+                        return disabled_file.read()
+                                .then(data => {
+                                    return {
+                                        backend: "chronyd",
+                                        enabled: false,
+                                        servers: parse_servers(data)
+                                    };
+                                });
+                    }
+                });
+    }
+
+    function set_custom_ntp_chronyd(config) {
+        const enabled_file = cockpit.file(chronyd_sources_enabled, { superuser: "try" });
+        const disabled_file = cockpit.file(chronyd_sources_disabled, { superuser: "try" });
+
+        const text = "# This file is automatically generated by Cockpit\n\n" + config.servers.map(s => `server ${s}\n`).join("");
+
+        if (config.enabled)
+            return enabled_file.replace(text).then(() => disabled_file.replace(null));
+        else
+            return disabled_file.replace(text).then(() => enabled_file.replace(null));
+    }
+
+    self.get_custom_ntp = function () {
+        return get_timesync_backend().then(backend => {
+            if (backend == "timesyncd")
+                return get_custom_ntp_timesyncd();
+            else if (backend == "chronyd")
+                return get_custom_ntp_chronyd();
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18298-20230207-124146-95ee9c44-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)